pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -100,6 +119,56 @@ func validateResourceGroup(client *Client, ic *types.InstallConfig, path *field.
 	return append(allErrs, field.NotFound(path.Child("resourceGroupID"), ic.AlibabaCloud.ResourceGroupID))
 }
 
+func validateVpc(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	vpcs, err := client.ListVpcs(ic.AlibabaCloud.VpcID)
+	if err != nil {
+		return append(allErrs, field.InternalError(path.Child("vpcID"), err))
+	}
+	if vpcs.TotalCount == 1 {
+		return allErrs
+	}
+	return append(allErrs, field.NotFound(path.Child("vpcID"), ic.AlibabaCloud.VpcID))
+}
+
+func validateVSwitchs(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.AlibabaCloud.VpcID == "" {
+		return append(allErrs, field.Required(path.Child("vswitchIDs"), "Use the existing VSwitchs, vpcID must be specified"))
+	}
+
+	for _, id := range ic.AlibabaCloud.VSwitchIDs {
+		vswitchs, err := client.ListVSwitchs(id)
+		if err != nil {
+			return append(allErrs, field.InternalError(path.Child("vswitchIDs"), err))
+		}
+		if vswitchs.TotalCount != 1 {
+			allErrs = append(allErrs, field.NotFound(path.Child("vswitchIDs"), id))
+			continue
+		}
+		if vswitchs.VSwitches.VSwitch[0].VpcId != ic.AlibabaCloud.VpcID {
+			allErrs = append(allErrs, field.Invalid(path.Child("vswitchIDs"), id, fmt.Sprintf("The VSwitch %s does not belong to vpc %s", id, ic.AlibabaCloud.VpcID)))
```

Comment:
Do not include the ID of the vswitch in the details. That is already included in the field error.